pkg/utils/host/host.go

```diff
@@ -142,11 +131,17 @@ func AddFlags(command *cobra.Command) {
 		false,
 		"Automatically run in a privileged systemd unit if lacking enough capabilities",
 	)
+
+	// Enable the mount workaround by default when running inside a container.
+	automountFilesystemsDefault := false
+	if HostRoot != "" {
+		automountFilesystemsDefault = true
+	}
 	command.PersistentFlags().BoolVarP(
 		&autoMountFilesystemsFlag,
 		"auto-mount-filesystems",
 		"",
-		false,
+		automountFilesystemsDefault,
```

Comment:
I am wondering if it's ok to change the default value dynamically. Would `--help` display a different default value depending on `$HOST_ROOT`?

What do you think about changing the default value to true unconditionally? I am not sure what's the best, so you can choose either way.

Reply:
I prefer to change the default value. I can be confusing for the users to have a flag that doesn't have any effect when `$HOST_ROOT` is defined.

Reply:
When one runs `ig --help`, it doesn't show the default value of bool variables:

```bash
$ ./ig --help
Collection of gadgets for containers
      [...]
      --auto-mount-filesystems          Automatically mount bpffs, debugfs and tracefs if they are not already mounted
      [...]
```

I suppose it is because it's expected to be always false and only true if I use such a flag. So, we have to manually add it within the description to make it visible.

Another option is to make it string and set it as "auto" by default:

```bash
      --mount-filesystems          Mount bpffs, debugfs and tracefs if they are not already mounted. Valid values are: true, false, auto (default "auto").
```

Reply:
Do we really need to change this more? I think just changing the default value is good enough for the purpose of this PR: To make it easier to run ig on containers.

Reply:
IMO, at least the default value should be present when one runs `ig --help`.